From 80a61cd7c71a989478ee50ee1a30c2df623811aa Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 22 Apr 2013 18:36:29 +0300
Subject: [PORT FROM MAIN] Telephony: fix javacrash in extracting mcc

BZ: 60137

Currenlty, operatorNumeric.length is not checked
before extracting mcc from operatorNumeric
resulting in StringIndexOutOfBoundsException.

Check the operatorNumeric length before extracting
mcc.

Change-Id: I843dcd8e8bb0c8bfcbab4d5bac93aafe2611cef1
Orig-Change-Id: I0462ead8fd531e3fa57b870d52998bee0570442c
Signed-off-by: Jun Wu <junx.wu@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/gsm/GsmServiceStateTracker.java      |    8 +++-----
 1 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index d4d2edc..12addcd 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -903,8 +903,8 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
             operatorNumeric = mSS.getOperatorNumeric();
             mPhone.setSystemProperty(TelephonyProperties.PROPERTY_OPERATOR_NUMERIC, operatorNumeric);
 
-            if (operatorNumeric == null) {
-                if (DBG) log("operatorNumeric is null");
+            if (operatorNumeric == null || operatorNumeric.length() < 3) {
+                if (DBG) log("operatorNumeric is Invalid");
                 mPhone.setSystemProperty(TelephonyProperties.PROPERTY_OPERATOR_ISO_COUNTRY, "");
                 mGotCountryCode = false;
                 mNitzUpdatedTime = false;
@@ -915,9 +915,7 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
                     mcc = operatorNumeric.substring(0, 3);
                     iso = MccTable.countryCodeForMcc(Integer.parseInt(mcc));
                 } catch ( NumberFormatException ex){
-                    loge("pollStateDone: countryCodeForMcc error" + ex);
-                } catch ( StringIndexOutOfBoundsException ex) {
-                    loge("pollStateDone: countryCodeForMcc error" + ex);
+                    loge("pollStateDone: Mcc Invalid(not a number)" + ex);
                 }
 
                 mPhone.setSystemProperty(TelephonyProperties.PROPERTY_OPERATOR_ISO_COUNTRY, iso);
-- 
1.7.4.1

